From 643cdee055388d040d125daacebd2a1c4be9d712 Mon Sep 17 00:00:00 2001
From: Les Lee <lesl@google.com>
Date: Tue, 16 Aug 2022 16:07:04 +0800
Subject: [PATCH] wifi: Reset to default SAP configuration when doing factory
 reset

Also fixed incorrect the isUserConfiguration value when doing reset to default
SAP configuration.

Bug: 241927115
Test: atest -c FrameworksWifiTests
Change-Id: Ia212666450fd9fe3a95393defa2fb56383bab83a
(cherry picked from commit df96a4f2945bda0df602f3f0bd53bc21b8d5b37f)
Merged-in: Ia212666450fd9fe3a95393defa2fb56383bab83a
(cherry picked from commit 0c3de17f513b94fc053f31f05b1b3160b25f8b35)
Merged-In: Ia212666450fd9fe3a95393defa2fb56383bab83a
---
 .../com/android/server/wifi/WifiApConfigStore.java    | 11 ++++-------
 .../java/com/android/server/wifi/WifiServiceImpl.java |  3 +++
 .../android/server/wifi/WifiApConfigStoreTest.java    |  2 +-
 .../com/android/server/wifi/WifiServiceImplTest.java  |  1 +
 4 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/service/java/com/android/server/wifi/WifiApConfigStore.java b/service/java/com/android/server/wifi/WifiApConfigStore.java
index 3144bbac0..70659c12f 100644
--- a/service/java/com/android/server/wifi/WifiApConfigStore.java
+++ b/service/java/com/android/server/wifi/WifiApConfigStore.java
@@ -177,14 +177,11 @@ public class WifiApConfigStore {
      * and the main Wifi thread (CMD_START_AP).
      */
     public synchronized void setApConfiguration(SoftApConfiguration config) {
-        if (config == null) {
-            config = getDefaultApConfiguration();
-        } else {
-            config = sanitizePersistentApConfig(config);
-        }
+        SoftApConfiguration newConfig = config == null ? getDefaultApConfiguration()
+                : new SoftApConfiguration.Builder(sanitizePersistentApConfig(config))
+                        .setUserConfiguration(true).build();
         persistConfigAndTriggerBackupManagerProxy(
-                new SoftApConfiguration.Builder(updatePersistentRandomizedMacAddress(config))
-                    .setUserConfiguration(true).build());
+                updatePersistentRandomizedMacAddress(newConfig));
     }
 
     /**
diff --git a/service/java/com/android/server/wifi/WifiServiceImpl.java b/service/java/com/android/server/wifi/WifiServiceImpl.java
index d89ed4a47..4101d1494 100644
--- a/service/java/com/android/server/wifi/WifiServiceImpl.java
+++ b/service/java/com/android/server/wifi/WifiServiceImpl.java
@@ -5024,6 +5024,9 @@ public class WifiServiceImpl extends BaseWifiService {
             removePasspointConfigurationInternal(null, config.getUniqueId());
         }
         mWifiThreadRunner.post(() -> {
+            EventLog.writeEvent(0x534e4554, "241927115", -1,
+                    "Reset SoftApConfiguration to default configuration");
+            mWifiApConfigStore.setApConfiguration(null);
             mPasspointManager.clearAnqpRequestsAndFlushCache();
             mWifiConfigManager.clearUserTemporarilyDisabledList();
             mWifiConfigManager.removeAllEphemeralOrPasspointConfiguredNetworks();
diff --git a/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java b/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java
index 087b719b5..69fafec1c 100644
--- a/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java
+++ b/service/tests/wifitests/src/com/android/server/wifi/WifiApConfigStoreTest.java
@@ -326,7 +326,7 @@ public class WifiApConfigStoreTest extends WifiBaseTest {
         verifyDefaultApConfig(mDataStoreSource.toSerialize(), TEST_DEFAULT_AP_SSID);
         verify(mWifiConfigManager).saveToStore(true);
         verify(mBackupManagerProxy).notifyDataChanged();
-        assertTrue(store.getApConfiguration().isUserConfigurationInternal());
+        assertFalse(store.getApConfiguration().isUserConfigurationInternal());
     }
 
     /**
diff --git a/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java b/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
index 6d1b2c0bd..5159fb8e6 100644
--- a/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
+++ b/service/tests/wifitests/src/com/android/server/wifi/WifiServiceImplTest.java
@@ -6154,6 +6154,7 @@ public class WifiServiceImplTest extends WifiBaseTest {
         // Let the final post inside the |factoryReset| method run to completion.
         mLooper.dispatchAll();
 
+        verify(mWifiApConfigStore).setApConfiguration(null);
         verify(mWifiConfigManager).removeNetwork(
                 openNetwork.networkId, Binder.getCallingUid(), TEST_PACKAGE_NAME);
         verify(mWifiConfigManager).removeNetwork(
-- 
2.39.0.rc1.256.g54fd8350bd-goog

